<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CamGrab</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true">background-color: rgb(0, 0, 0);</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="GLVideoWidget" name="cameraViewFinder" native="true">
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionToggleCamera"/>
   <addaction name="separator"/>
   <addaction name="actionGrabImage"/>
   <addaction name="actionOpenOutputDirectory"/>
   <addaction name="actionToggleSettings"/>
   <addaction name="actionToggleFilters"/>
   <addaction name="actionToggleConsole"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="consoleDockWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Console</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTextEdit" name="consoleOutput">
       <property name="font">
        <font>
         <family>DejaVu Sans Mono</family>
         <pointsize>7</pointsize>
        </font>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="settingsDockWidget">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>91</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="0">
      <widget class="QLineEdit" name="outDirEdit">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="outDirButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>42</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="outDirLabel">
       <property name="text">
        <string>Output directory</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="filtersDockWidget">
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>144</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Filters</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QCheckBox" name="enableFilters">
       <property name="text">
        <string>Enable</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="filtersFrame">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3"/>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionToggleCamera">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/switch.png</normaloff>:/resources/images/switch.png</iconset>
   </property>
   <property name="text">
    <string>Start camera</string>
   </property>
  </action>
  <action name="actionOpenOutputDirectory">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/folder_open.png</normaloff>:/resources/images/folder_open.png</iconset>
   </property>
   <property name="text">
    <string>Open output directory</string>
   </property>
   <property name="toolTip">
    <string>Open output directory</string>
   </property>
  </action>
  <action name="actionGrabImage">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/photo.png</normaloff>:/resources/images/photo.png</iconset>
   </property>
   <property name="text">
    <string>Grab image</string>
   </property>
   <property name="toolTip">
    <string>Grab image</string>
   </property>
  </action>
  <action name="actionRecordMovie">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/movie.png</normaloff>:/resources/images/movie.png</iconset>
   </property>
   <property name="text">
    <string>Record movie</string>
   </property>
   <property name="toolTip">
    <string>Record movie</string>
   </property>
  </action>
  <action name="actionToggleSettings">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/cogs.png</normaloff>:/resources/images/cogs.png</iconset>
   </property>
   <property name="text">
    <string>Open settings</string>
   </property>
   <property name="toolTip">
    <string>Open settings</string>
   </property>
  </action>
  <action name="actionToggleConsole">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/shell.png</normaloff>:/resources/images/shell.png</iconset>
   </property>
   <property name="text">
    <string>Toggle console</string>
   </property>
   <property name="toolTip">
    <string>Toggle console</string>
   </property>
  </action>
  <action name="actionToggleFilters">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/resources/images/glasses.png</normaloff>:/resources/images/glasses.png</iconset>
   </property>
   <property name="text">
    <string>Filters</string>
   </property>
   <property name="toolTip">
    <string>Filters</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLVideoWidget</class>
   <extends>QWidget</extends>
   <header>glvideowidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionToggleCamera</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleCamera(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleConsole</sender>
   <signal>toggled(bool)</signal>
   <receiver>consoleDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>463</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>consoleDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleConsole</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>463</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleCamera</sender>
   <signal>triggered(bool)</signal>
   <receiver>actionGrabImage</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleCamera</sender>
   <signal>triggered(bool)</signal>
   <receiver>actionRecordMovie</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>settingsDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleSettings</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleSettings</sender>
   <signal>toggled(bool)</signal>
   <receiver>settingsDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>204</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outDirButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>chooseOutputDirectory()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenOutputDirectory</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>openOutputDirectory()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGrabImage</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>grabImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleFilters</sender>
   <signal>toggled(bool)</signal>
   <receiver>filtersDockWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filtersDockWidget</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionToggleFilters</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>enableFilters</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>enableFilters(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggleCamera(bool)</slot>
  <slot>chooseOutputDirectory()</slot>
  <slot>openOutputDirectory()</slot>
  <slot>grabImage()</slot>
  <slot>enableFilters(bool)</slot>
 </slots>
</ui>
